assets/js/modules/analytics-4/datastore/report.js

```diff
@@ -90,7 +90,7 @@ const fetchGetReportStore = createFetchStore( {
 		);
 		invariant(
 			isValidMetrics( metrics ),
-			'metrics for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property.'
+			'metrics for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property. Metric names must match the expression ^[a-zA-Z0-9_]+$.'
```

Comment:
Note, the proposed message _"metrics for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property matching the expression `^[a-zA-Z0-9_]+$`."_ is a little inaccurate as it seems to suggest that _only_ object name properties will be validated against the regular expression, whereas _all_ metric names will be, whether passed as a string or an object. I've tweaked the message to reflect this.